src/stx/common/trex_rx_timesync.cpp

```diff
@@ -8,89 +8,92 @@
 
 void RXTimesync::handle_pkt(const rte_mbuf_t *m, int port) {
     if (m_timesync_engine->getTimesyncMethod() == TimesyncMethod::PTP) {
-        uint16_t rx_tstamp_idx = 0;
 
-        if (hardware_timestamping_enabled) {
-            rx_tstamp_idx = m->timesync;
-        }
+        timespec t { 0, 0 };
 
         if (m->pkt_len < ETH_HDR_LEN)
             return;
 
-        uint64_t m_timestamp = 0;
-        if (CGlobalInfo::m_options.is_timesync_rx_callback_enabled()) {
-            m_timestamp = m->timestamp;
+        if (hardware_timestamping_enabled && (m->ol_flags & PKT_RX_IEEE1588_TMST)) {
+            if (rte_eth_timesync_read_rx_timestamp(port, &t, m->timesync) != 0) {
+                printf("Cannot read hardware timestamp from hardware");
+            }
+        } else if (CGlobalInfo::m_options.is_timesync_rx_callback_enabled()) {
+            t = timestampToTimespec(m->timestamp);
+        } else {
+            if (clock_gettime(CLOCK_REALTIME, &t) != 0) {
+                printf("Something is wrong!!! Cannot read time from kernel!");
+            }
```

Comment:
I'm not sure if it is ~all right to try~ necessary to get the timestamp on every incoming PTP packet.  I mean the code looks cleaner and all, but there is some performance drawback.

Reply:
we don't have that much PTP packets so it's ok for me to timestamp all of them

Reply:
AFAIR TRex would try to handle every incoming packet with this method when running with `--software` mode

Reply:
As we talked, if this is true then all this code need to be changed as it is not optimized.